datasets/electricity/electricity.py

```diff
@@ -0,0 +1,217 @@
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URL = "https://archive.ics.uci.edu/ml/machine-learning-databases/00321/LD2011_2014.txt.zip"
+
+
+class Electricty(datasets.GeneratorBasedBuilder):
+    """Hourly electricity consumption of 370 points/clients."""
+
+    VERSION = datasets.Version("1.0.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="uci",
+            version=VERSION,
+            description="Original UCI time series.",
+        ),
+        datasets.BuilderConfig(
+            name="lstnet",
+            version=VERSION,
+            description="Electricity time series preporcessed as in LSTNet paper.",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "lstnet"
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "start": datasets.Value("string"),
+                "target": datasets.Sequence(datasets.Value("float32")),
+                "feat_static_cat": datasets.Sequence(datasets.Value("uint64")),
+                # "feat_static_real":  datasets.Sequence(datasets.Value("float32")),
+                # "feat_dynamic_real":  datasets.Sequence(datasets.Sequence(datasets.Value("uint64"))),
+                # "feat_dynamic_cat": datasets.Sequence(datasets.Sequence(datasets.Value("uint64"))),
+                "item_id": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features, uncomment supervised_keys line below and
+            # specify them. They'll be used if as_supervised=True in builder.as_dataset.
+            # supervised_keys=("sentence", "label"),
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLS
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        data_dir = dl_manager.download_and_extract(_URL)
+
+        # define the prediction problem # TODO save these in metadata
+        freq = "1H"
+        prediction_length = 24
+        rolling_evaluations = 7
```

Comment:
It makes sense to add these variables to the config as attributes to support custom values from the user.
```python
class ElectrictyConfig(datasets.BuilderConfig):
    """[docstring for attributes]"""  
    freq: str = "1H"
    prediction_length: int = 24
    rolling_evaluations: int = 7

...
BUILDER_CONFIGS = [
        ElectrictyConfig(
            name="uci",
            version=VERSION,
            description="Original UCI time series.",
        ),
        ElectrictyConfig(
            name="lstnet",
            version=VERSION,
            description="Electricity time series preporcessed as in LSTNet paper.",
        ),
    ]
```
Then, you can access the values as `self.config.{attr}` in the script (e.g. `self.config.freq`)

We should also add a section to the README where we explain each attribute in more detail (we can also add sample calls to `load_dataset` with these attributes, etc.).

Reply:
ah perfect! that makes a lot of sense I believe

Reply:
Let's add a new section `Dataset Usage` to the card. There you can explain the meaning of the config parameters `freq`, `prediction_length` and `rolling_evaluations` in a format similar to Dataset Fields and add an example call  (e.g. `load_dataset("electricity", "uci", rolling_evaluations=..., ...)`.